controllers/operator.go

```diff
@@ -395,6 +400,22 @@ func (o *operator) updateDeploy(deploy *appsv1.Deployment) (*appsv1.Deployment,
 	return deploy, nil
 }
 
+func (o *operator) managePodHealth(cl client.Client, conditions ...string) error {
+	podList := corev1.PodList{}
+	cl.List(context.Background(), &podList, client.MatchingLabels(map[string]string{KonsumeratorManagerAnnotation : "true"}))
+	for _, pod := range podList.Items {
+		if pod.Status.Phase == corev1.PodUnknown {
+			if err := cl.Delete(context.Background(), &pod); err != nil {
```

Comment:
the same goes for the call above (cl.List)

Reply:
good point. I have set a deadline of `30s` currently. I don't know if that's too much. Was mostly thinking about a deadline of `100ms` - but in a worst case scenario I don't know how long it will take to process 1000's of pods.